source/Irrlicht/CBAWMeshFileLoader.cpp

```diff
@@ -210,19 +210,20 @@ ICPUMesh* CBAWMeshFileLoader::createMesh(io::IReadFile* _file)
 	const uint32_t filesize = _file->getSize();
 	uint8_t* const fileBuffer = (uint8_t*)malloc(filesize);
 	_file->read(fileBuffer, filesize);
-	core::BawFileV1* const bawFile = (core::BawFileV1*)fileBuffer;
-	const uint32_t HEADERS_FILE_OFFSET = sizeof(core::BawFileV1::fileHeader) + sizeof(core::BawFileV1::numOfInternalBlobs) + sizeof(core::BawFileV1::blobOffsets[0])*bawFile->numOfInternalBlobs;
+	const core::BawFileV1* const bawFile = (core::BawFileV1*)fileBuffer;
+	const uint32_t HEADERS_FILE_OFFSET = bawFile->calcHeadersOffset();
+	const uint32_t BLOBS_FILE_OFFSET = bawFile->calcBlobsOffset();
 
 	SContext ctx;
 	ctx.filePath = _file->getFileName();
 	if (ctx.filePath[ctx.filePath.size()-1] != '/')
 		ctx.filePath += "/";
 	std::map<uint64_t, SBlobData>::iterator meshBlobData;
 
-	core::BlobHeaderV1* const headers = (core::BlobHeaderV1*)(fileBuffer + HEADERS_FILE_OFFSET);
+	const core::BlobHeaderV1* const headers = (core::BlobHeaderV1*)(fileBuffer + HEADERS_FILE_OFFSET);
```

Comment:
Just load all headers at the start, then find your mesh to load (we may support pulling named meshes or textures out of big archives).